src/analyzer/protocol/modbus/events.bif

```diff
@@ -294,3 +294,42 @@ event modbus_read_fifo_queue_request%(c: connection, headers: ModbusHeaders, sta
 ## fifos: The register values read from the FIFO queue on the device.
 event modbus_read_fifo_queue_response%(c: connection, headers: ModbusHeaders, fifos: ModbusRegisters%);
 
+## Generated for a Modbus Diagnostics request.
+##
+## c: The connection.
+##
+## headers: The headers for the modbus function.
+##
+## diag: The diagnostics data, including subfunction-specific values.
+event modbus_diagnostics_request%(c: connection, headers: ModbusHeaders, diag: ModbusDiagnostics%);
```

Comment:
I can see the motivation for the `ModbusDiagnostics` record for extensible, but could also see a more primitive one instead not doing the conversion of `data` to stick closer to what the protocol is. Should be possible to build conversion above.
```
modbus_diagnostics_request(c: connection, headers, subfunction: count, data: string);
modbus_diagnostics_response(c: connection, headers, subfunction: count, data: string)
```

Then in script-land can render subfunction into a string representation (via const table) and convert `data` to count via `bytestring_to_count` or delimiter if wanted given subfunction.

Reply:
Yeah, that's what I had originally in the first commit, and switched to the record version later. It's easy enough to revert the second commit.

Reply:
Reverted the second commit.